src/io/xsd/simulation.cpp

```diff
@@ -0,0 +1,2717 @@
+// C++ data binding compiler.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 2 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+//
+// In addition, as a special exception, Code Synthesis gives permission
+// to link this program with the Xerces-C++ library (or with modified
+// versions of Xerces-C++ that use the same license as Xerces-C++), and
+// distribute linked combinations including the two. You must obey the GNU
+// General Public License version 2 in all respects for all of the code
+// used other than Xerces-C++. If you modify this copy of the program, you
+// may extend this exception to your version of the program, but you are
+// not obligated to do so. If you do not wish to do so, delete this
+// exception statement from your version.
+//
+// Furthermore, Code Synthesis makes a special exception for the Free/Libre
+// and Open Source Software (FLOSS) which is described in the accompanying
+// FLOSSE file.
+//
+
+// Begin prologue.
+//
+//
+// End prologue.
+
+#include <xsd/cxx/pre.hxx>
+
+#include "simulation.h"
+
+// doubleVec_t
+//
+
+const doubleVec_t::x_type& doubleVec_t::x() const
+{
+    return this->x_.get();
+}
+
+doubleVec_t::x_type& doubleVec_t::x()
+{
+    return this->x_.get();
+}
+
+void doubleVec_t::x(const x_type& x)
+{
+    this->x_.set(x);
+}
+
+const doubleVec_t::y_type& doubleVec_t::y() const
+{
+    return this->y_.get();
+}
+
+doubleVec_t::y_type& doubleVec_t::y()
+{
+    return this->y_.get();
+}
+
+void doubleVec_t::y(const y_type& x)
+{
+    this->y_.set(x);
+}
+
+const doubleVec_t::z_type& doubleVec_t::z() const
+{
+    return this->z_.get();
+}
+
+doubleVec_t::z_type& doubleVec_t::z()
+{
+    return this->z_.get();
+}
+
+void doubleVec_t::z(const z_type& x)
+{
+    this->z_.set(x);
+}
+
+// intVec_t
+//
+
+const intVec_t::x_type& intVec_t::x() const
+{
+    return this->x_.get();
+}
+
+intVec_t::x_type& intVec_t::x()
+{
+    return this->x_.get();
+}
+
+void intVec_t::x(const x_type& x)
+{
+    this->x_.set(x);
+}
+
+const intVec_t::y_type& intVec_t::y() const
+{
+    return this->y_.get();
+}
+
+intVec_t::y_type& intVec_t::y()
+{
+    return this->y_.get();
+}
+
+void intVec_t::y(const y_type& x)
+{
+    this->y_.set(x);
+}
+
+const intVec_t::z_type& intVec_t::z() const
+{
+    return this->z_.get();
+}
+
+intVec_t::z_type& intVec_t::z()
+{
+    return this->z_.get();
+}
+
+void intVec_t::z(const z_type& x)
```

Comment:
I think the _t here is more of a suffix to denote that this is a type rather than indicating a snake case name. I think using snake case for the whole name would make the suffix less visible, but you are right that this styling is not that consistent. I will look out for this for future namings, leaving this for now so that we do not need to rewrite a lot of input files. Hope this is ok!